samples/snippets/src/test/java/pubsub/Retry.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pubsub;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+@Retention(RetentionPolicy.RUNTIME)
```

Comment:
@BenWhitehead could we use the junit rules you wrote here?

Reply:
junit5 has a @RepeatedTest annotation if we can upgrade?

Reply:
Sorry, let me clarify here :) @BenWhitehead worked on some shared JUnit rules in java-core, wondering if we can make use of them - https://github.com/googleapis/java-core/pull/327/files. @anguillanneuf let me know if these are a good fit for what you're trying to accomplish, otherwise feel free to ignore my suggestion :)

Reply:
https://github.com/googleapis/java-core/blob/master/google-cloud-core/src/test/java/com/google/cloud/testing/junit4/MultipleAttemptsRule.java is what I need, but it's in `src/test/`, is that something I can import?

Do you know if we can use the this for testing?
```xml
 <dependency>
    <groupId>org.junit.jupiter</groupId>
    <artifactId>junit-jupiter-engine</artifactId>
    <version>5.7.1</version>
    <scope>test</scope>
</dependency>
```